File: activator/src/main/java/my/sample/activator/Logger.java
package my.sample.activator;

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public interface Logger {

    /**
     * debug
     *
     * @param message the message to be logged
     */
    void debug(String message);

    /**
     * info
     *
     * @param message the message to be logged
     */
    void info(String message);

    /**
     * warn
     *
     * @param message the message to be logged
     */
    void warn(String message);

    /**
     * error
     *
     * @param message the message to be logged
     */
    void error(String message);
}

File: activator/src/main/kotlin/my/sample/activator/Logger.kt
package my.sample.activator

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
interface Logger {
    /**
     * debug
     *
     * @param message the message to be logged
     */
    fun debug(message: String)

    /**
     * info
     *
     * @param message the message to be logged
     */
    fun info(message: String)

    /**
     * warn
     *
     * @param message the message to be logged
     */
    fun warn(message: String)

    /**
     * error
     *
     * @param message the message to be logged
     */
    fun error(message: String)
}
